benchmarks/benchmarks/ci/esmf_regridder.py

```diff
@@ -363,3 +319,82 @@ def time_lazy_regridding(self, cache):
     def time_regridding_realisation(self, cache):
         assert self.result.has_lazy_data()
         _ = self.result.data
+
+
+class TimeRegridderIO(MultiGridCompare):
+    def setup(self, type):
```

Comment:
I know this is still WIP so I'll hold off a full review. But I think we need to start watching our 'setup hygiene'.

As we know, `setup()` runs multiple times per benchmark. So we should be thorough in making sure the only stuff in `setup()` is whatever is required to give us a 'fresh situation' ahead of running the benchmark(s). Two suggestions:
1. Move stuff that only needs running once into `setup_cache()`.
  (Since this can only pass objects to `setup()` via files and pickles, you will get a further speed up by removing the 'reload' steps of synthetic data generation - example below<sup>1</sup>).
2. Avoid running `setup()` steps that aren't going to be used. E.g. setting up `self.grid_to_mesh_regridder` for the `time_save_mesh_to_grid()` benchmark. Could either:
   * Split into separate classes
   * Further parameterise this class so that different setup steps are run depending on the active parameter.

If we're not careful with this the run times will get too long and we'll be forced to back out CI benchmarking and come up with something less convenient like overnight checking. It looks like you might already be experiencing this - 30mins for the last run! @abooton regularly said that 7mins was already too long 😳

---

<sup>1</sup>
https://github.com/SciTools-incubator/iris-esmf-regrid/blob/a23e97583cf8d27ae85d830d808b1bccdc074ac8/benchmarks/benchmarks/generate_data.py#L164-L166